### DISCUSSION TITLE
[Feature] When scanning a wallet, import existing labels.

### DISCUSSION THREAD

**Discussion by mutatrum** [cryptoadvance/specter-desktop#1267] [General]
When scanning a wallet, any labels that are already present on addresses or UTXOs should be imported into Specter.

For existing labels, a decision has to be made what the ground truth of a label is, the one in Specter, or the one in the bitcoin wallet. My first hunch would be the one in the bitcoin wallet, but more thought needs to go into this.

Per @k9ert: It enables that several specters can share the same Bitcoin-Node.

**Comment by k9ert** [comment]
> Per @k9ert: It enables that several specters can share the same Bitcoin-Node.

I don't think this is a good idea. What's the use-case of sharing a wallet? In fact we're even creating a specific random ID which is used so that different specter are not interfering even IF they would use the same wallet-names.

So in fact you can't currently "import" an existing wallet from Bitcoin-Core. However, Ben is thinking about MOVING a wallet from one core instance to another. But t i guess that wouldn't help here at all. 

Flagging this as an enhancement for now.
